OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -69,26 +71,78 @@ public override Activity Tick(Actor self)
 			if (attackStatus.HasFlag(AttackStatus.NeedsToMove))
 				return moveActivity;
 
+			if (attackStatus.HasFlag(AttackStatus.FrozenSwitch))
+			{
+				// When shooting a non-frozen target which became frozen,
+				// this makes sure that the target line will remain even after
+				// the underlying actor is destroyed
+				self.SetTargetLine(frozenSwitchActivity.Target, Color.Red);
```

Comment:
> The case where I'm describing is the case where the target died but we keep shooting its frozen image since we do not know it died.

That's what I'm talking about as well. It doesn't work in this PR.

Testcase:
Let an artillery fire on a frozen actor. Then sell (or destroy) the real actor. The arty will still have the red target line but won't shoot anymore.